devlib/utils/android.py

```diff
@@ -177,7 +177,12 @@ def newline_separator(self):
     # versions of the ls tool in Android pre-v7.
     def _setup_ls(self):
         command = "shell '(ls -1); echo \"\n$?\"'"
-        output = adb_command(self.device, command, timeout=self.timeout)
+        try:
+            output = adb_command(self.device, command, timeout=self.timeout)
+        except subprocess.CalledProcessError as e:
+            raise HostError(
```

Comment:
Hm, good point. On second though, you're right. Given that this is the first thing that happens for non-IP devices, the assumptions are essentially the same as in `adb_get_device` case. Hm may be there should be a "ConnectionError" or something in both cases. Anyway, for now, consistency with `adb_get_device` makes sense. I withdraw my previous comment.